packages/amplication-server/src/core/auth/auth.service.ts

```diff
@@ -423,6 +539,125 @@ export class AuthService {
     return workspace as unknown as Workspace & { users: AuthUser[] };
   }
 
+  private async createPreviewWorkspaceWithProjectAndResource(account: Account) {
+    const workspaceName = `Amplication-${this.generateRandomString()}`;
+    const projectName = account.previewAccountType;
+
+    const workspace = (await this.workspaceService.createPreviewWorkspace(
+      {
+        data: {
+          name: workspaceName,
+        },
+        include: WORKSPACE_INCLUDE,
+      },
+      account.id
+    )) as unknown as Workspace & { users: AuthUser[] };
+    const [user] = workspace.users as AuthUser[];
+
+    const project = await this.projectService.createProject(
+      {
+        data: {
+          name: projectName,
+          workspace: {
+            connect: {
+              id: workspace.id,
+            },
+          },
+        },
+      },
+      user.id
+    );
+
+    let resource: Resource;
+    switch (account.previewAccountType) {
+      case PreviewAccountType.BreakingTheMonolith:
+        resource = await this.createBreakingTheMonolithPreviewService(
+          project.id,
+          user
+        );
+        break;
+      case PreviewAccountType.None:
+        throw new AmplicationError(
+          `${PreviewAccountType.None} is not a preview account, but a regular account`
+        );
+      default:
+        throw new AmplicationError(
+          `Unsupported preview account type: ${account.previewAccountType}`
+        );
+    }
+
+    return {
+      user,
+      workspace,
+      project,
+      resource,
+    };
+  }
+
+  private async createBreakingTheMonolithPreviewService(
+    projectId: string,
+    user: User
+  ): Promise<Resource> {
+    const serviceName = "Monolith";
+    const createPreviewServiceArgs = this.createPreviewServiceArgs({
+      projectId: projectId,
+      name: serviceName,
+      description: "Monolith Service",
+      adminUIPath: `./apps/${serviceName}-admin`,
+      serverPath: `./apps/${serviceName}`,
+      generateAdminUI: true,
+      generateGraphQL: true,
+      generateRestApi: true,
+    });
+
+    const resource = await this.resourceService.createPreviewService({
+      args: createPreviewServiceArgs,
+      user,
+      nonDefaultPluginsToInstall: [],
+      requireAuthenticationEntity: false,
+    });
+
+    return resource;
+  }
+
+  private createPreviewServiceArgs({
+    name,
+    description,
+    adminUIPath,
+    serverPath,
+    generateAdminUI,
+    generateGraphQL,
+    generateRestApi,
+    authEntityName = "User",
+    projectId,
+  }: CreatePreviewServiceArgs) {
+    return {
+      data: {
+        name,
+        description,
+        resourceType: EnumResourceType.Service,
+        project: {
+          connect: {
+            id: projectId,
+          },
+        },
+        serviceSettings: {
+          authProvider: EnumAuthProviderType.Jwt,
+          authEntityName,
```

Comment:
didn't we say we want the preview account to be created with no auth and no User entity?